call su(1) with login argument 'root'

sg2342 opened the pull request:
Omission of the login argument in sudo_run_command will make the call to su(1)
fail on FreeBSD, NetBSD, DragonFLy, Darwin and any other UNIX where the su(1)
executable is not a GNU coreutils (derived) su.
Either because the '-c' option before the login argument defines a 'login class'
or because the '-c' option is not defined at all.

su root -c 'the command string to be executed'

will pass '-c' to the spawned shell and work on every platform.

fixes ocaml/opam-depext#51 (however newer MacOS installations tend
to have no root password set)

sg2342 commented:
i re-based this on 2.0

rjbou commented:
Thanks for the contribution!
lgtm, but can you rebase and clean your branch on master? There is a little problem with the git history, it show several already present commits.

sg2342 committed:
call su(1) with login argument 'root'

Omission of the login argument in sudo_run_command will make the call to su(1)
fail on FreeBSD, NetBSD, DragonFLy, Darwin and any other UNIX where the su(1)
executable is not a GNU coreutils (derived) su.
Either because the '-c' option before the login argument defines a 'login class'
or because the '-c' option is not defined at all.

su root -c 'the command string to be executed'

will pass '-c' to the spawned shell and work on every platform.

fixes ocaml/opam-depext#51 (however newer MacOS installations tend
to have no root password set)

sg2342 commented:
rebased on master...

rjbou commented:
Thanks!
